### PR TITLE
Move evm logic to evm utils

### PR DESCRIPTION
**Pull request by ErnoW**
---
name: 'Pull request'
about: A new pull request
---

### Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [x] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [x] I have made corresponding changes to the documentation
- [x] I have updated Typescript definitions when needed

### Issue Description

- Moved all Evm dataTypes to `evm-utils`
- Moved register of config of `formatEvmAddress` and `formatEvmChainId` to `evm-utils`
- Moved register of config of `defaultEvmApiChain` to `evm-api`

Other fixes/improvements
- Naming of `setupEvmApi`
- Config of eslint was broken (syntax error)
- Fixed versioning of newly created packages
- cache format function in nx
- Fix prettier versioning as devDependency
- Fix building of test-node demo app by using symLinked packages when building

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -29,10 +29,8 @@
     "typescript": "^4.5.5"
   },
   "dependencies": {
-    "@ethersproject/address": "^5.6.0",
     "@ethersproject/bignumber": "^5.6.0",
     "@ethersproject/units": "^5.6.0",
```
Comment (b4rtaz) [inline packages/core/package.json:33]: Do we need these dependencies?

File: packages/evmUtils/src/config/EvmUtilsConfigSetup.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { EvmUtilsConfig } from './EvmUtilsConfig';
+
+export class EvmUtilsConfigSetup {
+  public static register(config: Config) {
+    if (!config.hasKey(EvmUtilsConfig.formatEvmAddress)) {
+      config.registerKey(EvmUtilsConfig.formatEvmAddress);
+    }
+
+    if (!config.hasKey(EvmUtilsConfig.formatEvmChainId)) {
+      config.registerKey(EvmUtilsConfig.formatEvmChainId);
```
Comment (b4rtaz) [inline packages/evmUtils/src/config/EvmUtilsConfigSetup.ts:11]: I think this approach is wrong. This approach hides modules usage errors. You should get an error, if you use the module system in wrong way.

File: packages/evmUtils/src/test/setup.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { MoralisCoreProvider } from '@moralisweb3/core';
+import { MoralisEvmUtils } from '../MoralisEvmUtils';
+
+export const setupEvm = () => {
```
Comment (b4rtaz) [inline packages/evmUtils/src/test/setup.ts:4]: Should be `setupEvmUtils()`.

**Review by b4rtaz — CHANGES_REQUESTED** [review]
Just 3 remarks. Good job.

File: packages/core/package.json (hunk 1 above)
Comment (ErnoW) [reply packages/core/package.json:33]: Only as long as it is being used by the Sol datatypes in core.

**Review by b4rtaz — APPROVED** [review]

**Pull request merged**